drivers/dma/dma_andes_atcdmac100.c

```diff
@@ -0,0 +1,533 @@
+		(((struct dma_atcdmac100_cfg *)dev->config)->base + 0x50 + DMA_CH_OFFSET(ch))
+#define DMA_CH_LL_PTR_L(dev, ch)	\
+		(((struct dma_atcdmac100_cfg *)dev->config)->base + 0x54 + DMA_CH_OFFSET(ch))
+
+/* DMA Control Register Bits */
+#define DMA_CTRL_RESET_OFFSET	0
+#define DMA_CTRL_RESET_MASK		GENMASK(0, 0)
+
+/* Source burst size options */
+#define DMA_BSIZE_1			(0)
+#define DMA_BSIZE_2			(1)
+#define DMA_BSIZE_4			(2)
+#define DMA_BSIZE_8			(3)
+#define DMA_BSIZE_16		(4)
+#define DMA_BSIZE_32		(5)
+#define DMA_BSIZE_64		(6)
+#define DMA_BSIZE_128		(7)
+
+/* Source/Destination transfer width options */
+#define DMA_WIDTH_BYTE		(0)
+#define DMA_WIDTH_HALFWORD	(1)
+#define DMA_WIDTH_WORD		(2)
+#define DMA_WIDTH_RESERVED	(3)
+
+/* DMA Channel Control Register Definition */
+#define DMA_CH_CTRL_SRCREQB5_MASK		BIT(31)
+#define DMA_CH_CTRL_DSTREQB5_MASK		BIT(30)
+#define DMA_CH_CTRL_PRIORITY_HIGH		BIT(29)
+#define DMA_CH_CTRL_SBSIZE_MASK			GENMASK(24, 22)
+#define DMA_CH_CTRL_SBSIZE(n)			FIELD_PREP(DMA_CH_CTRL_SBSIZE_MASK, (n))
+#define DMA_CH_CTRL_SWIDTH_MASK			GENMASK(21, 20)
+#define DMA_CH_CTRL_SWIDTH(n)			FIELD_PREP(DMA_CH_CTRL_SWIDTH_MASK, (n))
+#define DMA_CH_CTRL_DWIDTH_MASK			GENMASK(19, 18)
+#define DMA_CH_CTRL_DWIDTH(n)			FIELD_PREP(DMA_CH_CTRL_DWIDTH_MASK, (n))
+#define DMA_CH_CTRL_SMODE_HANDSHAKE		BIT(17)
+#define DMA_CH_CTRL_DMODE_HANDSHAKE		BIT(16)
+#define DMA_CH_CTRL_SRCADDRCTRL_MASK	GENMASK(15, 14)
+#define DMA_CH_CTRL_SRCADDR_INC			FIELD_PREP(DMA_CH_CTRL_SRCADDRCTRL_MASK, (0))
+#define DMA_CH_CTRL_SRCADDR_DEC			FIELD_PREP(DMA_CH_CTRL_SRCADDRCTRL_MASK, (1))
+#define DMA_CH_CTRL_SRCADDR_FIX			FIELD_PREP(DMA_CH_CTRL_SRCADDRCTRL_MASK, (2))
+#define DMA_CH_CTRL_DSTADDRCTRL_MASK	GENMASK(13, 12)
+#define DMA_CH_CTRL_DSTADDR_INC			FIELD_PREP(DMA_CH_CTRL_DSTADDRCTRL_MASK, (0))
+#define DMA_CH_CTRL_DSTADDR_DEC			FIELD_PREP(DMA_CH_CTRL_DSTADDRCTRL_MASK, (1))
+#define DMA_CH_CTRL_DSTADDR_FIX			FIELD_PREP(DMA_CH_CTRL_DSTADDRCTRL_MASK, (2))
+#define DMA_CH_CTRL_SRCREQ_MASK			GENMASK(11, 8)
+#define DMA_CH_CTRL_SRCREQ(n)			FIELD_PREP(DMA_CH_CTRL_SRCREQ_MASK, (n))
+#define DMA_CH_CTRL_DSTREQ_MASK			GENMASK(7, 4)
+#define DMA_CH_CTRL_DSTREQ(n)			FIELD_PREP(DMA_CH_CTRL_DSTREQ_MASK, (n))
+#define DMA_CH_CTRL_INTABT				BIT(3)
+#define DMA_CH_CTRL_INTERR				BIT(2)
+#define DMA_CH_CTRL_INTTC				BIT(1)
+#define DMA_CH_CTRL_ENABLE				BIT(0)
+
+/* DMA Interrupt Status Register Definition */
+#define	DMA_INT_STATUS_TC_MASK		GENMASK(23, 16)
+#define	DMA_INT_STATUS_ABORT_MASK	GENMASK(15, 8)
+#define	DMA_INT_STATUS_ERROR_MASK	GENMASK(7, 0)
+#define DMA_INT_STATUS_TC_VAL(x)	FIELD_GET(DMA_INT_STATUS_TC_MASK, (x))
+#define DMA_INT_STATUS_ABORT_VAL(x)	FIELD_GET(DMA_INT_STATUS_ABORT_MASK, (x))
+#define DMA_INT_STATUS_ERROR_VAL(x)	FIELD_GET(DMA_INT_STATUS_ERROR_MASK, (x))
+#define DMA_INT_STATUS_CH_MSK(ch)	(0x111 << ch)
+
+typedef void (*atcdmac100_cfg_func_t)(void);
+
+struct __attribute__((packed, aligned(4))) chain_block {
+	uint32_t ctrl;
+	uint32_t srcaddrl;
```

Comment:
Hey, whats the 'l' at the end of srcaddr?

Reply:
in ATCDMAC300 we have low and high registers. I removed all the srcaddrh type variables and left srcaddrl type. May be i can rename it to remove this l part as well.